### DISCUSSION TITLE
How to expose a ui5-webcomponents function from custom webcomponents

### DISCUSSION THREAD

**Discussion by akshil-verma** [UI5/webcomponents#8755] [Q&A] [answered]
Hi,

We have created a custom webcomponent based on ui5-webcomponent's framework. Our web component is utilized in a frontend project that is built using the Luigi micro frontend framework. Until version 1.18.0, everything worked fine, but from version 1.19.0 the frontend project has been facing theming-related issues, i.e., the theme changes only partially for the whole frontend project. We realized that this problem is due to the new CSS runtime fix in https://github.com/SAP/ui5-webcomponents/pull/7703 and this problem is solved when the frontend framework calls the `setTheme` function.

However, the frontend framework team insists on not importing any of the ui5-webcomponents packages as they are not directly dependent on it. So, my question is if there's any recommended way of exposing the `setTheme` function from within a custom webcomponent?

Best regards,
Akshil Verma

**Comment by petyabegovska** [comment]
Hi @akshil-verma,

Can you confirm the exact version of UI5 Web Components? Are you using the scoping feature: https://github.com/SAP/ui5-webcomponents/blob/main/docs/2-advanced/03-scoping.md?

Best regards,
Petya

**Reply by akshil-verma** [reply]
We are currently on 1.18.0 and are planning to upgrade to the latest version. We haven't used the scoping feature.

**Comment by petyabegovska** [comment]
And one more question: Do you have a custom theme created with the Theme designer?

**Reply by akshil-verma** [reply]
No, this issue is for the existing themes. For example, switching for horizon theme to horizon dark.

**Comment by akshil-verma** [comment]
Hi,

Any leads on this?

**Comment by akshil-verma** [comment, marked as answer]
Closing this discussion because of the solution provided in https://github.com/SAP/ui5-webcomponents/issues/8900
